tree-sitter-topas/grammar.js

```diff
@@ -347,6 +347,29 @@ module.exports = grammar({
       field('argument', $.identifier),
     ),
 
+    // ------- Keywords -------- //
+
+    _keyword_statement: $ => choice(
+      $.variable_declaration,
+      $.variable_assignment,
+    ),
+
+    variable_declaration: $ => seq(
+      field('keyword', choice('prm', 'local')),
+      field('name', refineable($.identifier)),
+      field('value', choice($._literal, $.simple_assignment)),
+    ),
+
+    variable_assignment: $ => seq(
+      field('keyword', 'existing_prm'),
+      field('name', refineable($.identifier)),
+      field('value', choice(
+        $.simple_assignment,
+        $.compound_assignment,
+        $._literal,
```

Comment:
Does `$._literal` here mean `existing_prm foo 4` is allowed? I don't see any tests for such a case

Reply:
Good spot, I didn't really think about that. For every other keyword that would be allowed, but I can't find any examples of for `existing_prm` that don't use an equation. I don't have access to TOPAS at the moment, else I'd test it.

The technical document marks the `existing_prm` keyword with `E`, suggesting that it accepts equation or literal, so I think for now I'll add a test, then, if necessary, I can remove it once I've confirmed one way or the other. Does that sound reasonable?

Reply:
Do you have access to TOPAS again? If so, probably worth testing before we merge

Reply:
Tested it out - literals alone aren't a valid input for `existing_prm`, nor `#prm`, so I've removed that option from the grammar and the tests. Because why would this language be self-consistent? :roll_eyes: